pkg/reports/error.go

```diff
@@ -0,0 +1,140 @@
+/*******************************************************************************
+*
+* Copyright 2022 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package reports
+
+import (
+	"database/sql"
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/sapcc/limes/pkg/core"
+	"github.com/sapcc/limes/pkg/db"
+)
+
+var scrapeErrorsQuery = db.SimplifyWhitespaceInSQL(`
+	SELECT d.uuid, d.name, p.uuid, p.name, ps.type, ps.checked_at, ps.scrape_error_message
+	  FROM projects p
+	  LEFT OUTER JOIN domains d ON d.id = p.domain_id
+	  LEFT OUTER JOIN project_services ps ON ps.project_id = p.id
+	WHERE %s GROUP BY d.uuid, d.name, p.uuid, p.name, ps.type, ps.checked_at, ps.scrape_error_message
+	HAVING ps.scrape_error_message != ''
```

Comment:
- `LEFT OUTER JOIN` can be replaced by `JOIN` since the `ps.scrape_error_message != ''` condition makes it impossible to have partially-null result rows.
- `GROUP BY` is not necessary since no aggregate functions are used and the query will never produce multiple rows for one project service (with basically the same justification as in the previous point).
- 
```suggestion
	  JOIN domains d ON d.id = p.domain_id
	  JOIN project_services ps ON ps.project_id = p.id
	WHERE %s AND ps.scrape_error_message != ''
```